src/crypto/crypto_context.cc

```diff
@@ -190,18 +197,90 @@ int SSL_CTX_use_certificate_chain(SSL_CTX* ctx,
 
 }  // namespace
 
+void ReadSystemStoreCertificates(
+    std::vector<std::string>* system_root_certificates) {
+#ifdef _WIN32
+  const HCERTSTORE hStore = CertOpenSystemStoreW(0, L"ROOT");
+  CHECK_NE(hStore, nullptr);
+
+  auto cleanup =
+      OnScopeLeave([hStore]() { CHECK_EQ(CertCloseStore(hStore, 0), TRUE); });
```

Comment:
As an alternative approach, consider implementing something like a `CertStorePointer` in the same general idea as the various `**Pointer` types defined in `crypto_util.h`. Something along the lines of...

```cpp
struct CertStorePointer {
  const HCERTSTORE store_;
  CertStorePointer() : store_(CertOpenSystemStoreW(0, L"ROOT")) {}
  ~CertStorePointer() { CHECK_EQ(CertCloseStore(store_, 0), TRUE); }
};
```

Reply:
Thanks for the suggestion, I addressed the same. PTAL

Reply:
@jasnell I tried addressing this PR comment in this commit: https://github.com/nodejs/node/commit/84932230ae4cbd0897bbab2b214bc57000f9d931 
which ended up in broken windows build: https://github.com/nodejs/node/actions/runs/3375763877/jobs/5602718800
I've no idea what is leading to this broken behavior for the builds, can you take a look?